src/html-conventions-lib/general-connectors-html-convention.xsl

```diff
@@ -688,4 +688,133 @@
+                    else
+                        ()
+                    
+                    "
+            />
+        </xsl:if>
+        <xsl:if test="boolean($targetRole)">
+            <xsl:variable name="elementsFound"
+                select="f:getElementByName($targetRole, root($connector))"/>
+            <xsl:variable name="attributesFound"
+                select="f:getAttributeByName($targetRole, root($connector))"/>
+
+            <xsl:sequence
+                select="
+                    if (count($elementsFound) > 1 or count($attributesFound) > 0) then
+                        f:generateHtmlError(fn:concat('The connector role name ', $targetRole, ' is not unique.'))
+                    else
+                        ()
+                    
+                    "
+            />
+        </xsl:if>
+    </xsl:template>
+
+
+    <xd:doc>
+        <xd:desc>[association-dependency-checker] - Connector role names are used in other
+            connectors </xd:desc>
+        <xd:param name="connector"/>
+        <xd:param name="isDependency"/>
+    </xd:doc>
+    <xsl:template name="connectorRoleCheck">
+        <xsl:param name="isDependency"/>
+        <xsl:param name="connector"/>
+        <xsl:variable name="sourceRole" select="$connector/source/role/@name"/>
+        <xsl:variable name="targetRole" select="$connector/target/role/@name"/>
+
+
+        <xsl:if test="boolean($sourceRole)">
+            <xsl:choose>
+                <xsl:when test="$isDependency = fn:true()">
+                    <xsl:variable name="connectorsFound"
+                        select="f:getConnectorByName($sourceRole, root($connector))/properties[@ea_type = 'Association']"/>
+
+                    <xsl:sequence
+                        select="
+                            if (count($connectorsFound) > 0) then
+                                f:generateHtmlError(fn:concat('The connector role name ', $sourceRole, ' is used in other connectors.'))
+                            else
+                                ()
+                            
+                            "
+                    />
+                </xsl:when>
+                <xsl:otherwise>
+                    <xsl:variable name="connectorsFound"
+                        select="f:getConnectorByName($sourceRole, root($connector))/properties[@ea_type = 'Dependency']"/>
+
+                    <xsl:sequence
+                        select="
+                            if (count($connectorsFound) > 0) then
+                                f:generateHtmlError(fn:concat('The connector role name ', $sourceRole, ' is used in other connectors.'))
+                            else
+                                ()
+                            
+                            "
+                    />
+                </xsl:otherwise>
+            </xsl:choose>
+        </xsl:if>
+
+        <xsl:if test="boolean($targetRole)">
```

Comment:
what is the difference from the logic above? 
same/similar code seems to be copied.

Reply:
source vs target? Cmon that should have been a function called twice instead of the function code copy-pasted. such code unmaintainable. This is how bugs emerge.